src/gradescopeapi/api/api.py

```diff
@@ -1,200 +1,359 @@
+        )
+
+
+@app.post("/assignments/update_dates")
+def update_assignment_dates(
+    assignment_dates: AssignmentDates,
+    session: requests.Session = Depends(get_gs_connection_session),
+):
+    """
+    Update the release and due dates for an assignment.
 
+    Args:
+        assignment_dates (AssignmentDates): Pydantic model containing the course_id, assignment_id, release_date, due_date, and late_due_date.
+        session (requests.Session, optional): The session object used for making HTTP requests.
+            Defaults to the session instance provided by the get_session dependency.
 
-def update_student_late_due_date(
-    token: TokenModel, student_id: str, course_id: str, assignment_id: str
-) -> bool:
-    """_summary_
+    Returns:
+        dict: A dictionary with a "message" key indicating if the assignment dates were updated successfully.
+
+    Raises:
+        HTTPException: If the assignment dates update fails, with a 400 Bad Request status code and the error message "Failed to update assignment dates".
+    """
+    try:
+        success = update_assignment_date(
+            session=session,
+            course_id=assignment_dates.course_id,
+            assignment_id=assignment_dates.assignment_id,
+            release_date=assignment_dates.release_date,
+            due_date=assignment_dates.due_date,
+            late_due_date=assignment_dates.late_due_date,
+        )
+        if success:
+            return {
+                "message": "Assignment dates updated successfully",
+                "status_code": status.HTTP_200_OK,
+            }
+        else:
+            raise HTTPException(
+                status_code=400, detail="Failed to update assignment dates"
+            )
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+
+@app.post("/assignments/extensions", response_model=dict)
+def get_assignment_extensions(
+    extension_data: ExtensionData,
+    session: requests.Session = Depends(get_gs_connection_session),
+):
+    """
+    Get all extensions for an assignment.
 
     Args:
-        token (TokenModel): _description_
-        student_id (str): _description_
-        assignment_id (str): _description_
+        extension_data (ExtensionData): Pydantic model containing the course_id and assignment_id.
+        session (requests.Session, optional): The session object used for making HTTP requests.
+            Defaults to the session instance provided by the get_session dependency.
 
     Returns:
-        bool: if updating the student late due date was successful, returns True
-              otherwise returns False
+        dict: A dictionary containing the extensions, where the keys are user IDs and the values are Extension objects.
+
+    Raises:
+        HTTPException: If the request to get extensions fails, with a 500 Internal Server Error status code and the error message.
     """
```

Comment:
NitPick : Could we think of other exceptions possible here

Reply:
will add other exceptions to all functions in FastAPI